### PR TITLE
Improve import updates with slice change plans and port deltas

### PR DESCRIPTION
**Pull request by astaticvoid**
*Issue #, if available:*
#101 

*Description of changes:*
Multiple improvements to service import logic.
* EndpointSlice change plan calculates necessary changes for endpoints.
* Derived service ports are updated based on imported external changes.
* Internal Port model is simplified and used broadly for comparisons ignoring order.
* Constructor logic moved to util class
* Some common test struct factory logic moved to controllers_common_test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by bendu — DISMISSED** [review]

File: pkg/controllers/utils.go [hunk 1]
```diff
@@ -1,91 +1,213 @@
...
-			}
-		}
-		if !match {
+	aMap := make(map[string]*model.Port)
+	for _, portA := range a {
+		aMap[portA.GetID()] = portA
+	}
+
+	for _, portB := range b {
+		if !portB.Equals(aMap[portB.GetID()]) {
```
Comment (runakash) [inline pkg/controllers/utils.go:119]: should we add check if `portB.GetID()` key is not found in the `aMap`

File: pkg/controllers/endpointslice_plan.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	maxEndpointsPerSlice int
+
+	// Service to reconcile endpoints in
+	Service *v1.Service
+
+	// ServiceImportName name used to create new EndpointSlices
+	ServiceImportName string
+
+	// Current EndpontSlices
+	Current []discovery.EndpointSlice
```
Comment (runakash) [inline pkg/controllers/endpointslice_plan.go:33]: Could be optimal to use list of pointers here so we don't copy the complete list.

**Review by runakash — APPROVED** [review]
Overall looks good. Few minor comments.

File: pkg/controllers/utils.go (hunk 1 above)
Comment (astaticvoid) [reply pkg/controllers/utils.go:119]: Good catch

File: pkg/controllers/endpointslice_plan.go (hunk 2 above)
Comment (astaticvoid) [reply pkg/controllers/endpointslice_plan.go:33]: That's fair.

File: pkg/controllers/common_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
+package controllers
```
[the remark this answers is not in the document]
Comment (astaticvoid) [reply pkg/controllers/common_test.go]: renamed to `controllers_common_test.go`

**Review by bendu — APPROVED** [review]

**Review by runakash — APPROVED** [review]

**Pull request merged**